signingscript/src/signingscript/sign.py

```diff
@@ -860,12 +861,15 @@ def get_hawk_header(url, user, password, content_type, content_hash):
 
 
 @time_async_function
-async def call_autograph(session, url, user, password, sign_req):
+async def call_autograph(session, url, user, password, sign_req, skip_encoding=False):
     """Call autograph and return the json response."""
     content_type = "application/json"
 
     request_body = tempfile.TemporaryFile("w+b")
-    write_signing_req_to_disk(request_body, sign_req)
+    if skip_encoding:
+        request_body.write(sign_req)
+    else:
+        write_signing_req_to_disk(request_body, sign_req)
```

Comment:
Looks like `make_signing_req` just returns a non encoded dictionary. If `make_files_signing_req` also returned this, would we be able to drop this `skip_encoding` parameter?

Reply:
I'd have to change the code in write_signing_req_to_disk. [It's writing it to the file assuming we are using one of the single file endpoints] (https://github.com/mozilla-releng/scriptworker-scripts/blob/master/signingscript/src/signingscript/sign.py#L814)

Reply:
Makes sense. As per my earlier comment I think that's the way to go.. But even if we don't go that route, it feels like we should make a duplicate `write_files_signing_req_to_disk` function rather than have your `make_files_signing_req` doing all the JSON building.

Reply:
I added a `write_files_signing_req_to_disk` function, I think this is more consistent with the rest of the code